client/src/pages/Suggest/index.jsx

```diff
@@ -13,6 +14,9 @@ export default function Suggest() {
 	const resourceService = useService(ResourceService);
 	const topicService = useService(TopicService);
 
+	const principal = usePrincipal();
+	const isAdmin = principal ? principal.name === "Cyf Admin" : false;
```

Comment:
❌ This is **not** how you determine if a user is an administrator. That user (as you saw when you first tried to log in) may or may not actually be set up as an admin, depending on what's in the DB. There may be zero, one or many admins, and in the real system there won't be any user with that name. Have a look at the other components that behave conditionally based on the user type to see how it should be done.

💡 In modern JS you can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to make this kind of thing neater:

```suggestion
	const isAdmin = principal?.name === "Cyf Admin";
```

Reply:
Implementation changed as `const isAdmin = principal?.is_admin;`